(fix|COS-407): Improve HTML content rendering by cleaning and formatting - fix bug in email preview ui

Kasia9090 opened the pull request:
## Changes

The changes are introduced to the "HtmlContentRenderer.tsx" to improve the rendering of HTML content. Broadly, there are two key changes associated with this commit:

1. Clean HTML function: A new function "cleanHTML" is created to efficiently clean the HTML content. This includes removing unnecessary tags like "!doctype", "html", "title", "body", "meta" and the content within style tags for optimal rendering.

2. The handling of HTML attributes is improved to exclude ‘style’ attributes while preserving others — aiding in better content presentation.

3. Content coming to html content renderer is now sanitized to improve security of the app 

Overall, these changes provide better UI/UX for log entries while ensuring smoother HTML content rendering.

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

## Additional context

Kasia9090 committed:
(fix|COS-407): Improve HTML content rendering by cleaning and formatting - fix bug in email preview ui

The changes are introduced to the "HtmlContentRenderer.tsx" to improve the rendering of HTML content. Broadly, there are two key changes associated with this commit:

1. Clean HTML function: A new function "cleanHTML" is created to efficiently clean the HTML content. This includes removing unnecessary tags like "!doctype", "html", "title", "body", "meta" and the content within style tags for optimal rendering.

2. The handling of HTML attributes is improved to exclude ‘style’ attributes while preserving others — aiding in better content presentation.

Overall, these changes provide better UI/UX for log entries while ensuring smoother HTML content rendering.

acalinica commented on packages/apps/spaces/ui/presentation/HtmlContentRenderer/HtmlContentRenderer.tsx:
```diff
@@ -15,7 +15,19 @@
+  const tagsToRemove = ['!doctype', 'html', 'title', 'body', 'meta'];
+  let outputHTML = inputHTML;
 
+  outputHTML = outputHTML.replace(/<style[^>]*>.*<\/style>/gm, '');
```
should we address this?

Kasia9090 replied:
fixed

Kasia9090 committed:
Refactor HTML sanitization logic in HtmlContentRenderer

This commit replaces the custom HTML sanitization function with the sanitize-html library in HtmlContentRenderer.tsx. The custom function was removing certain tags from the HTML content but it was not sufficient for all cases and caused UI issues. The sanitize-html library provides a more robust and secure implementation. It is configured to allow specific tags and attributes that we accept in our HTML content, while sanitizing the rest.

Kasia9090 committed:
Updated styling properties in HtmlContentRenderer

This commit updates the padding, display, border, and borderRadius properties in the HtmlContentRenderer component. The padding has been increased to improve the readability and aesthetics of the content. The display was set to 'block' to fix a layout issue where content was not properly aligned. The border and borderRadius were added to make the content more distinct and visually appealing. A redundant div styling was removed to simplify the styling rules.
